src/main/java/org/kaazing/netx/ws/internal/WsURLConnectionImpl.java

```diff
@@ -435,13 +435,27 @@ public void doFail(int code, String exceptionMessage) throws IOException {
         throw new IOException(exceptionMessage);
     }
 
-    public WebSocketContext getContext(WebSocketExtensionSpi sentinel, boolean reverse) {
+    public DefaultWebSocketContext getIncomingContext() {
```

Comment:
I can. But, I will have to downcast in WsInputStream, WsReader, and WsMessageReader to retrieve the sentinel extension to add a terminal consumer. Note that this is not a public API.